packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/quanticSearchBoxInput.js

```diff
@@ -185,8 +188,11 @@ export default class QuanticSearchBoxInput extends LightningElement {
     if (this.areSuggestionsOpen && selectedSuggestion) {
       this.sendSelectSuggestionEvent(selectedSuggestion.rawValue);
     } else {
-      this.sendSubmitSearchEvent();
+      if (!this.isClearButtonTargeted) {
+        this.sendSubmitSearchEvent();
+      }
```

Comment:
According to my previous comments I didn't mean  to say to change anything with the logic, cause it was fine and it was giving the right behaviour. 

This last commit is changing the logic.

With the current logic, the suggestions list  will not be opened after clicking on the enter ket on the clear input icon because of line 212 that will be executed:
```
this.input.blur();
```

Reply:
I would suggest coming back to your previous solution with just a minor restructuring:

```
  handleEnter() {
    if (!this.isClearButtonTargeted) {
      const selectedSuggestion =
        this.suggestionListElement?.getCurrentSelectedValue();
      if (this.areSuggestionsOpen && selectedSuggestion) {
        this.sendSelectSuggestionEvent(selectedSuggestion.rawValue);
      } else {
        this.sendSubmitSearchEvent();
      }
      this.input.blur();
    }
  }
```
And
```
  onKeyup(event) {
    switch (event.key) {
      case keys.ENTER:
        this.handleEnter();
        break;
      case keys.ARROWUP:
        this.suggestionListElement?.selectionUp();
        break;
      case keys.ARROWDOWN:
        this.suggestionListElement?.selectionDown();
        break;
      default:
        this.handleKeyValues();
    }
    this.isClearButtonTargeted = false;
  }
```

Reply:
yeah makes sense, I saw it was still working then after the current version, but it might have not deployed properly or something